### PR TITLE
Fix bug where dates in the past or current date are ignored.

### PR DESCRIPTION
**Pull request by dhirajshettymoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/utils/VisitBalancesUtil.kt [hunk 1]
```diff
@@ -5,25 +5,36 @@ import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.pri
...
   fun calculateAvailableVos(visitBalance: VisitBalancesDto?): Int {
     return (visitBalance?.remainingVo ?: 0) + (visitBalance?.remainingPvo ?: 0)
   }
 
-  fun calculateVoRenewalDate(visitBalance: VisitBalancesDto?, dateFrom: LocalDate = LocalDate.now()): LocalDate {
-    val latestIepAdjustDate = calculateIepAdjustDate(visitBalance, dateFrom)
-    val latestPrivilegedIepAdjustDate = calculatePrivilegedIepAdjustDate(visitBalance, dateFrom)
+  fun calculateVoRenewalDate(visitBalance: VisitBalancesDto?): LocalDate {
+    val currentDate = dateUtil.getCurrentDate()
+    val latestIepAdjustDate = calculateIepAdjustDate(visitBalance, currentDate)
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/utils/VisitBalancesUtil.kt:15]: is this latestIepAdjustDate a fancy way of saying VO

**Review by adaviesMOJ — APPROVED** [review]
LGTM!

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/utils/VisitBalancesUtil.kt (hunk 1 above)
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/utils/VisitBalancesUtil.kt:15]: it is what comes from prison API but I could possibly rename that on our end to a name that suits us.

Comment (AledWEvans) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/utils/VisitBalancesUtil.kt:15]: so could we call it latestVoAdjustedDate or may be latestAdjustedDate

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/utils/VisitBalancesUtil.kt [hunk 2]
```diff
@@ -5,25 +5,36 @@ import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.pri
...
+    val latestPrivilegedIepAdjustDate = calculatePrivilegedIepAdjustDate(visitBalance, currentDate)
 
     return minOf(latestIepAdjustDate, latestPrivilegedIepAdjustDate)
   }
 
   private fun calculateIepAdjustDate(visitBalance: VisitBalancesDto?, dateFrom: LocalDate): LocalDate {
-    // IEP Adjust date is latestIepAdjustDate or if latestIepAdjustDate not available then calculated as TODAY + 14 days
-    return visitBalance?.latestIepAdjustDate ?: dateFrom.plusDays(14)
+    // IEP Adjust date is latestIepAdjustDate or if latestIepAdjustDate not available or today or in the past then calculated as TODAY + 14 days
+    val latestIepAdjustDate = getLatestAdjustDate(visitBalance?.latestIepAdjustDate, dateFrom)
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/utils/VisitBalancesUtil.kt:23]: same here

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/utils/VisitBalancesUtil.kt [hunk 3]
```diff
@@ -5,25 +5,36 @@ import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.pri
...
+    return latestIepAdjustDate ?: dateFrom.plusDays(14)
   }
 
   private fun calculatePrivilegedIepAdjustDate(visitBalance: VisitBalancesDto?, dateFrom: LocalDate): LocalDate {
-    // Privileged IEP Adjust date is latestPrivIepAdjustDate or if latestPrivIepAdjustDate not available then calculated as 1st day of next month
-    return visitBalance?.latestPrivIepAdjustDate ?: dateFrom.plusMonths(1).withDayOfMonth(1)
+    // Privileged IEP Adjust date is latestPrivIepAdjustDate or if latestPrivIepAdjustDate not available or today or in the past then calculated as 1st day of next month
+    getLatestAdjustDate(visitBalance?.latestIepAdjustDate, dateFrom)
+    val latestPrivIepAdjustDate = getLatestAdjustDate(visitBalance?.latestPrivIepAdjustDate, dateFrom)
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/utils/VisitBalancesUtil.kt:30]: latestPrivilegedAdustedDate I just hate Iep

**Review by AledWEvans — APPROVED** [review]
LGTM, just some comments on naming,.. your call

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/utils/VisitBalancesUtil.kt (hunk 1 above)
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/utils/VisitBalancesUtil.kt]: updated variable names

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/utils/VisitBalancesUtil.kt (hunk 2 above)
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/utils/VisitBalancesUtil.kt]: updated variable names

**Review by adaviesMOJ — APPROVED** [review]

**Pull request merged**